src/main/java/org/zalando/intellij/swagger/index/IndexFacade.java

```diff
@@ -10,24 +11,13 @@
 
 public class IndexFacade {
 
-  private final OpenApiIndexService openApiIndexService;
-  private final SwaggerIndexService swaggerIndexService;
-  private final DumbService dumbService;
-
-  public IndexFacade(
```

Comment:
Having a constructor with dependencies listed makes code much cleaner and readable rather than injecting the dependencies inside the class by using some sort of singleton (`ServiceManager` - I am not an expert in IDE plugins, but seems it is typical service registry).  

There is [IndexFacadeTest](src/test/java/org/zalando/intellij/swagger/index/IndexFacadeTest.java) below which illustrates this problem:

### Before refactoring

```java
private final IndexFacade indexFacade = new IndexFacade(fakeOpenApiIndexService, fakeSwaggerIndexService, dumbService);  
```

`IndexFacade` clearly states what kind of dependencies it needs.

### After refactoring

```java
private final IndexFacade indexFacade = new IndexFacade();
```

In this case it seems that `IndexFacade` doesn't have any dependencies, but this is not true. There are at least three services needed, but in order to understand this, developers have to jump inside this class and find them (if they know what to search for). And only after this it becomes clean why there are three mock services in the test.

Reply:
This was required as the it is enforced by the platform to have zero argument constructor.
```com.intellij.diagnostic.PluginException: Bean extension class constructor must not have parameters```